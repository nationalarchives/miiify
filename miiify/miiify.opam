# This file is generated by dune, edit dune-project instead
opam-version: "2.0"
synopsis: "A web annotation server"
description:
  "A simple web annotation server built from the same principles at Git"
maintainer: ["john.moore@nationalarchives.gov.uk"]
authors: ["john.moore@nationalarchives.gov.uk"]
license: "MIT"
tags: ["web" "annotation" "git" "irmin"]
homepage: "https://github.com/nationalarchives/miiify"
doc: "https://github.com/nationalarchives/miiify"
bug-reports: "https://github.com/nationalarchives/miiify/issues"
depends: [
  "dune" {>= "3.11"}
  "dream"
  "irmin-git"
  "irmin-pack"
  "odoc" {with-doc}
]
build: [
  ["dune" "subst"] {dev}
  [
    "dune"
    "build"
    "-p"
    name
    "-j"
    jobs
    "@install"
    "@runtest" {with-test}
    "@doc" {with-doc}
  ]
]
dev-repo: "git+https://github.com/nationalarchives/miiify.git"
