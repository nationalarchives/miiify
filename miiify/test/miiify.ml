open Miiify.Db
open Ezjsonm
open Lwt.Infix
open Lwt_io

let key = [ "k" ]
let j1 = dict [ ("test", string "hello world") ]
let j2 = dict [ ("test", string "hello world again") ]

let crud () =
  let ctx = create ~fname:"db" ~author:"miiify test" in
  add ~ctx ~key ~json:j1 ~message:"CREATE" >>= fun () ->
  count ~ctx ~key >>= fun n ->
  printf "%d\n" n >>= fun () ->
  get ~ctx ~key >>= fun j ->
  printf "%s\n" (value_to_string j) >>= fun () ->
  add ~ctx ~key ~json:j2 ~message:"UPDATE" >>= fun () ->
  count ~ctx ~key >>= fun n ->
  printf "%d\n" n >>= fun () ->
  get ~ctx ~key >>= fun j ->
  printf "%s\n" (value_to_string j) >>= fun () ->
  delete ~ctx ~key ~message:"DELETE" >>= fun () ->
  count ~ctx ~key >>= fun n -> printf "%d\n" n

let () = Lwt_main.run @@ crud ()
