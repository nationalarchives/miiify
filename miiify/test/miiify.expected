1
{"test":"hello world"}
1
{"test":"hello world again"}
0
